script.js

```diff
@@ -12,60 +12,169 @@ const confirmationButton = document.getElementById("confirmation-button");
+    itemSubtotal.textContent = `$${existingItem.price * existingItem.quantity}`;
+  } else {
+    selectedPack.quantity = 1;
+    cartItems.push({ ...selectedPack });
 
-const closeModal = () => {
-  confirmationModalOverlay.classList.add("opacity-0", "pointer-events-none");
-  confirmationModalOverlay.classList.remove("opacity-100");
+    const itemElement = document.createElement("div");
+    itemElement.id = `cart-item-${selectedPack.name
+      .replace(/\s+/g, "-")
+      .toLowerCase()}-${selectedPack.price}`;
 
-  darkOverlay.classList.add("opacity-0");
-  darkOverlay.classList.remove("opacity-60");
-};
+    itemElement.className =
+      "bg-primary w-full rounded-xl flex justify-start items-center p-4 gap-4";
+    itemElement.innerHTML = ` <div>
+          <img src="/assets/icons/${selectedPack.icon}" class="size-20" />
+        </div>
+        <div class="flex flex-col items-center gap-2">
+          <h3 class="text-black font-bold">${selectedPack.name}</h3>
+          <div class="flex flex-row items-center gap-2">
+            <img
+              src="/assets/icons/SubstractIcon.svg"
+              class="size-8 hover:scale-105 transition-all duration-300"
+            />
+            <p class="quantity-display">${selectedPack.quantity}</p>
+            <img
+              src="/assets/icons/AddIcon.svg"
+              class="size-8 hover:scale-105 transition-all duration-300"
+            />
+          </div>
+        </div>
+        <div
+          class="flex flex-col items-center text-base font-medium text-light-grey gap-2"
+        >
+          <p>$${selectedPack.price}</p>
+          <img
+            src="/assets/icons/BinIcon.svg"
+            class="size-8 hover:bg-light-orange rounded-full p-2"
+          />
+        </div>
+    `;
+    cartModal.appendChild(itemElement);
+
+    const summaryElement = document.createElement("div");
+    summaryElement.id = `summary-item-${selectedPack.name
+      .replace(/\s+/g, "-")
+      .toLowerCase()}-${selectedPack.price}`;
+
+    summaryElement.className = "flex justify-between text-sm font-semibold";
+    summaryElement.innerHTML = `<p class="text-light-grey">${
+      selectedPack.name
+    }</p><p class="subtotal-display">$${
+      selectedPack.price * selectedPack.quantity
+    }</p>`;
+
+    cartSummary.appendChild(summaryElement);
+  }
+
+  cartItemsNumber.textContent = cartItems.length;
+
+  updateTotalDisplay();
+
+  setTimeout(() => {
+    cartButton.classList.remove("scale-125");
+    cartItemsContainer.classList.replace("hidden", "flex");
+  }, 500);
+});
 
 closeModalButton.addEventListener("click", closeModal);
 cancelConfirmationButton.addEventListener("click", closeModal);
+
+cartButton.addEventListener("click", () => {
+  cartClicked = !cartClicked;
+
+  if (cartItems.length === 0) {
+    toggleModalVisibility(emptyCartModal, cartClicked);
+  } else {
+    toggleModalVisibility(cartModal, cartClicked);
+  }
```

Comment:
suggestion: you can use a ternary operator:
`const modalToToggle = cartItems.length === 0 ? emptyCartModal : cartModal;`
ant then have a single function call
`toggleModalVisibility(modalToToggle, cartClicked);`